plugins/ci/pkg/ci/ci.go

```diff
@@ -535,17 +537,100 @@ func ProcessHelmTemplates(helmConfigs []models.HelmConfig, tempFolder string, co
 			if err != nil {
 				return err
 			}
+		} else if helm.IsFluxFile() {
+			err := handleFluxHelmChart(helm, tempFolder, configFolder)
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("Could not determine the type of helm config.: %v", helm.Name)
 		}
 	}
 	return nil
 }
 
+func handleFluxHelmChart(helm models.HelmConfig, tempFolder string, configFolder string) error {
+	if helm.Name == "" || helm.Repo == "" {
+		return errors.New("Parameters 'name', 'repo' are required when using fluxFile")
+	}
+
+	fluxFile, err := os.Open(helm.FluxFile)
+	if err != nil {
+		return fmt.Errorf("Unable to open file %v: %v", helm.FluxFile, err)
+	}
+
+	fluxFileContent, err := ioutil.ReadAll(fluxFile)
+	if err != nil {
+		return fmt.Errorf("Unable to read file %v: %v", helm.FluxFile, err)
+	}
+
+	// Ideally we should use https://fluxcd.io/docs/components/helm/api/#helm.toolkit.fluxcd.io%2fv1
+	// However, the attempt was not possible due to being unable to parse `v1.Duration` successfully
+	type HelmReleaseModel struct {
+		Spec struct {
+			Chart struct {
+				Spec struct {
+					Chart   string `yaml:"chart"`
+					Version string `yaml:"version"`
+				} `yaml:"spec"`
+			} `yaml:"chart"`
+			Values     map[string]interface{} `yaml:"values"`
+			ValuesFrom []interface{}          `yaml:"valuesFrom"`
+		} `yaml:"spec"`
+	}
+
+	var helmRelease HelmReleaseModel
+	err = yaml.Unmarshal(fluxFileContent, &helmRelease)
+	if err != nil {
+		return fmt.Errorf("Unable to unmarshal file %v: %v", helm.FluxFile, err)
+	}
+
+	chartName := helmRelease.Spec.Chart.Spec.Chart
+	if chartName == "" {
+		return fmt.Errorf("Could not find required spec.chart.spec.chart in fluxFile %v", helm.FluxFile)
+	}
+
+	repoName := fmt.Sprintf("%s-%s-repo", helm.Name, helm.Chart)
+
+	err = util.RunCommand(exec.Command("helm", "repo", "add", repoName, helm.Repo), "Adding chart repository: "+repoName)
+	if err != nil {
+		return err
+	}
+
+	repoDownloadPath := fmt.Sprintf("%s/downloaded-charts/%s/", tempFolder, repoName)
+	chartDownloadPath := repoDownloadPath + chartName
+
+	chartFullName := fmt.Sprintf("%s/%s", repoName, chartName)
+	params := []string{"fetch", chartFullName, "--untar", "--destination", repoDownloadPath}
```

Comment:
Lots of this logic is duplicated. Can we extract the details we need (repo, chart name, version, etc) and pass those to a helper function?